book/website/communication/personal/personal-website.md

```diff
@@ -25,6 +25,56 @@ But this space is not necessarily limited to academic content; it can also be a
 This online presence can help researchers expand their visibility, engage with a wider audience, connect with potential collaborators, and even create new career opportunities.
 
+(cm-personal-website-basics)=
+# Basic elements of Web Development: Processes and Tools
+
+Web development is the entire process of creating and maintaining websites and web applications.
+
+It involves iterative ideation and implementation. The core phases associated with web development are analysis, design, development, testing & review, deployment, and maintenance.
+
+(cm-personal-website-analysis)=
+## Analysis 
+
+The goal of the analysis phase is to gain a deep understanding of the problem that a website is trying to solve so that the best solution can be found. By taking the time to carefully analyse the requirements of a website, people (researcher, web developer, programmer, and more) can ensure that the final product is effective and meets its users' needs.
+
+(cm-personal-website-design)=
+## Design
+
+In this phase, the website structure plus the visual components and content, including activity layouts, navigation panels, images, and videos, are added to provide a feature-rich audio-visual experience to the users. The design stage is crucial since it directly impacts the UI/UX of the website. It is important that the design of the website is responsive, highly intuitive, and easy to navigate.
+
+(cm-personal-website-development)=
+## Development
+
+The development (& coding) phase of the web development process is when the actual content (in case of low-code/no-code frameworks) and/or code (in case of high-code frameworks) for the website is written. For high-code framwworks, this can be done using a variety of programming languages, depending on the functionalities required in the website. This is where the developer takes all of the designs, wireframes, and other elements from the previous phases and turns them into a working website.
+
+In some cases, the entire site may be coded from scratch, but developers usually prefer using pre-existing tools or frameworks to speed up the process. Some of the commonly used frameworks are mentioned in the [Frameworks](cm-personal-website-frameworks) section below.
+
+(cm-personal-website-test-review)=
+## Testing and review
+
+The next stage in the process involves testing the website comprehensively to remove any bugs or errors before it goes live. This is a critical stage. There are different testing methods and techniques that are utilised to ensure that the website is performing as expected at optimum levels.
+
+The testing phase can be divided into two main categories:
+
+(cm-personal-website-test-functional)=
+### Functional testing: 
+Functional testing focuses on verifying that the website performs its intended tasks.
+
+(cm-personal-website-test-non-functional)=
+### Non-functional testing: 
+
+Non-functional testing assesses the website's performance, security, and usability.
```

Comment:
These sections are quite short and don't have much explanation.
It might be best here to link to another section of the book about testing?

Or maybe this could be a list with links to other resources?